_posts/2020-08-15-Named-Graphs.md

```diff
@@ -0,0 +1,539 @@
+individuals to each other using OWL's `sameAs` predicate.  Using this
+predicate to link all instances of an individual together has
+consequences for searching which has an effect on the use of Named
+Graphs and a large effect on the results returned.  Users must be
+aware of this as it can have a large effect on the results returned
+and the interpretation of the results returned.
+
+Stardog's documentation
+[states](https://www.stardog.com/docs/#_same_as_reasoning) that:
+
+> The way sameAs reasoning works differs from the OWL semantics
+> slightly in the sense that Stardog designates one canonical
+> individual for each sameAs equivalence set and only returns the
+> canonical individual. This avoids the combinatorial explosion in
+> query results while providing the data integration benefits.
+
+This means that if the user searches a Named Graph, if the canonical
+`owl:sameAs` is chosen such that it is in a different graph, it could
+appear in results in place of the one that is in the Named Graph being
+searched.  This can cause surprise to a user but they are technically
+and by definition the same; however, there is nothing that anyone can
+do about this due to the fact that Stardog does the above.
+Additionally, for an individual who is the same across many graphs,
+this can change each time the database is loaded as the system
+randomly assigns a canonical individual.
+
+GraphDB has an approach which is the exact opposite to Stardog.
+Instead of using forward-chaining to create `sameAs` relations,
+GraphDB creates an [equivalence
+class](https://www.w3.org/TR/owl2-syntax/#Equivalent_Classes) and does
+backwards-chaining:
+
+> There is no restriction on how to choose this single node that will represent the class as a whole, so we pick the first node that enters the class. After creating such a class, all statements with nodes from this class are altered to use the class representative. These statements also participate in the inference.
+>
+> The equivalence classes may grow when more owl:sameAs statements containing nodes from the class are added to the repository. Every time you add a new owl:sameAs statement linking two classes, they merge into a single class.
+>
+> During query evaluation, GraphDB uses a kind of backward chaining by enumerating equivalent URIs, thus guaranteeing the completeness of the inference and query results. It takes special care to ensure that this optimization does not hinder the ability to distinguish between explicit and implicit statements.
+
+More detail is available in GraphDB's
+[documentation](http://graphdb.ontotext.com/documentation/standard/sameas-optimisation.html).
+
+To demonstrate how this effects GraphDB's results, if the user runs the query:
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a
+from <http://example.com/LL>
+where {
+    ?a irishRel:nomName "Find"
+}
+```
+
+There will be 19 results as from Stardog.  
```

Comment:
Omit "as from Stardog". This implies that the above was run in Graph DB and returns the same number of results as when it is run in Stardog, whereas the first set of results are from Stardog.